src/zenml/integrations/label_studio/steps/label_studio_standard_steps.py

```diff
@@ -0,0 +1,272 @@
+) -> str:
+    """Gets preexisting dataset or creates a new one.
+
+    Args:
+        config: Step config.
+        context: Step context.
+
+    Returns:
+        The dataset name.
+
+    Raises:
+        TypeError: If you are trying to use it with an annotator that is not
+            Label Studio.
+        StackComponentInterfaceError: If no active annotator could be found.
+    """
+    annotator = context.stack.annotator  # type: ignore[union-attr]
+    from zenml.integrations.label_studio.annotators.label_studio_annotator import (
+        LabelStudioAnnotator,
+    )
+
+    if not isinstance(annotator, LabelStudioAnnotator):
+        raise TypeError(
+            "This step can only be used with the Label Studio annotator."
+        )
+    if annotator and annotator._connection_available():
+        preexisting_dataset_list = [
+            dataset
+            for dataset in annotator.get_datasets()
+            if dataset.get_params()["title"] == config.dataset_name
+        ]
+        if (
+            not preexisting_dataset_list
+            and annotator
+            and annotator._connection_available()
+        ):
+            registered_dataset = annotator.register_dataset_for_annotation(
+                config
+            )
+        elif preexisting_dataset_list:
+            return cast(str, preexisting_dataset_list[0].get_params()["title"])
+        else:
+            raise StackComponentInterfaceError("No active annotator.")
+
+        return cast(str, registered_dataset.get_params()["title"])
+    else:
+        raise StackComponentInterfaceError("No active annotator.")
+
+
+@step(enable_cache=False)
+def get_labeled_data(dataset_name: str, context: StepContext) -> List:  # type: ignore[type-arg]
+    """Gets labeled data from the dataset.
+
+    Args:
+        dataset_name: Name of the dataset.
+        context: The StepContext.
+
+    Returns:
+        List of labeled data.
+
+    Raises:
+        TypeError: If you are trying to use it with an annotator that is not
+            Label Studio.
+        StackComponentInterfaceError: If no active annotator could be found.
+    """
+    # TODO [MEDIUM]: have this check for new data *since the last time this step ran*
+    annotator = context.stack.annotator  # type: ignore[union-attr]
+    if not annotator:
+        raise StackComponentInterfaceError("No active annotator.")
+    from zenml.integrations.label_studio.annotators.label_studio_annotator import (
+        LabelStudioAnnotator,
+    )
+
+    if not isinstance(annotator, LabelStudioAnnotator):
+        raise TypeError(
+            "This step can only be used with the Label Studio annotator."
+        )
+    if annotator._connection_available():
+        dataset = annotator.get_dataset(dataset_name=dataset_name)
+        return dataset.get_labeled_tasks()  # type: ignore[no-any-return]
+    else:
+        raise StackComponentInterfaceError(
+            "Unable to connect to annotator stack component."
+        )
```

Comment:
```suggestion
    raise StackComponentInterfaceError(
        "Unable to connect to annotator stack component."
    )
```

Reply:
I always wonder if you don't lose something by doing it as you suggested: i.e. you lose the connection with the conditionals + the lineage that the error was raised because it failed these conditionals and not just because we got to the end of the method.